src/AspNet.Security.OAuth.Apple/AppleAuthenticationHandler.cs

```diff
@@ -137,19 +138,28 @@ protected override async Task<OAuthTokenResponse> ExchangeCodeAsync([NotNull] OA
         /// Extracts the claims from the token received from the token endpoint.
         /// </summary>
         /// <param name="token">The token to extract the claims from.</param>
+        /// <param name="identity">Represents a claims-based identity</param>
         /// <returns>
         /// An <see cref="IEnumerable{Claim}"/> containing the claims extracted from the token.
         /// </returns>
-        protected virtual IEnumerable<Claim> ExtractClaimsFromToken([NotNull] string token)
+        protected virtual IEnumerable<Claim> ExtractClaimsFromToken([NotNull] string token, [NotNull] ClaimsIdentity identity)
```

Comment:
This would be a breaking change to the API surface, so can't be part of a v3 release. Is there a way to refactor this in such a way as to not require a breaking change and still fix the issue?

Reply:
Good point.  I will work towards a different resolution.

Reply:
What are your thoughts about overloading this method?

Reply:
This is still a breaking change, but now functionally. The original method is no longer called, so if a caller had overridden it their customised implementation would no longer be called. To preserve that behaviour, you'd have to call the existing method from the new overload.

Reply:
Do you have a preference?  I could call the original method from the overloaded method as you suggest, or create a new method that is called right alongside, but not instead of, the original method?

Reply:
Is there a way to edit the logic in the existing method and put the new code before or after it's called? Then we don't need to add any new members and it just becomes an internal change in behaviour.

Reply:
Yes, we could add the logic to the existing "ExtractClaimsFromToken" method, however, without the ClaimsIdentity "identity" value, there is no way to check if the "Email" claim had already been added to the "identity" by the "ExtractClaimsFromUser" method.  So the "identity" could end up with a duplicate "Email" claim.

Reply:
I have the new overloaded method calling the original method at present.  That code change is pushed up.  You can see the dilemma and how the "identity" value is used to prevent the duplicate "Email" claim.  I enjoy the collaboration, so let me know if you see an alternative that you prefer.

Reply:
I created an alternative solution for comparison and consideration here: https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers/pull/409